### PR TITLE
Add search box to splash screen

### LINKED ISSUE #3540: Add search box to home page
**Requested By:** NMC

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #3540

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/HomePage/index.tsx [hunk 1]
```diff
@@ -20,13 +25,22 @@ const taxonTiles = f.store(() => (
 
 export function WelcomeView(): JSX.Element {
   const [mode] = userPreferences.use('welcomePage', 'general', 'mode');
+  const [query = ''] = useSearchParameter('q');
+  const value = React.useState(query);
+  const [pendingQuery] = value;
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/HomePage/index.tsx:30]: why not move these into the `<SearchForm />`?

`formId` is the only one that is needed outside of it

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]
good job refactoring that into a component and changing styles appropriately

have one suggestion though

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
**Can we make it this:**
image

**Instead of this:**
image

**Review by maxxxxxdlp — APPROVED** [review]

**Review by maxxxxxdlp — APPROVED** [review]

**Review by grantfitzsimmons — APPROVED** [review]

**Review by bronwyncombs — APPROVED** [review]
Looks good! Working as expected. 

Screenshot 2023-07-30 at 9 06 37 AM

full screen:
Screenshot 2023-07-30 at 9 07 59 AM

**Comment by bronwyncombs** [comment]
The search box is sized differently across browsers. Could be expected but if it's not: 

Screenshot 2023-07-29 at 11 25 47 PM

**Comment by carlosmbe** [comment]
Does it have a toggle? For example, if a user prefers have a minimalist layout. Can they turn it off? (I am that user)

**Comment by grantfitzsimmons** [comment]
I am OK with adding a pref, though it is not a priority.

File: specifyweb/frontend/js_src/lib/components/Preferences/UserDefinitions.tsx [hunk 2]
```diff
@@ -500,6 +500,14 @@ export const userPreferenceDefinitions = {
       general: {
         title: preferencesText.general(),
         items: {
+          addSearchBar: defineItem<boolean>({
+            title: preferencesText.addSearchBarHomePage(),
+            description: preferencesText.addSearchBarHomePage(),
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Preferences/UserDefinitions.tsx:505]: if description is the same as title, then why add description at all?
either specify a different description, or remove it

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

**Review by maxxxxxdlp — APPROVED** [review]

**Pull request merged**